test/test_dependencies.py

```diff
@@ -202,3 +204,34 @@ def test_dummy_skip_rosdep_multiple_keys_pkg(tmpdir):
     result = out_script.read_text()
     assert 'ros-dashing-ament-cmake' not in result
     assert 'ros-dashing-rclcpp' not in result
+
+
+@uses_docker
+def test_dependencies_stage_call(tmpdir):
```

Comment:
I don't think it's necessary for this to be a new test - I was thinking you could just add these modifications to `test_dummy_ros2_pkg` - since this is intended to be the new API to this functionality, that's a good place to start migrating the test code over